veps/sig-compute/vmtemplate-proposal.md

```diff
@@ -0,0 +1,752 @@
+              name: ${NAME}
+            name: disk-1
+          - cloudInitNoCloud:
+              userData: |-
+                #cloud-config
+                user: fedora
+                password: ${CLOUD_USER_PASSWORD}
+                chpasswd: { expire: False }
+            name: cloudinitdisk
+  parameters:
+  - name: NAME
+    description: VM name
+    from: 'fedora-[a-z0-9]{16}'
+    generate: expression
+  - name: INSTANCETYPE
+    description: VM instance type
+    value: u1.medium
+  - name: DATA_SOURCE_NAME
+    description: Name of the DataSource to clone
+    value: fedora
+  - name: DATA_SOURCE_NAMESPACE
+    description: Namespace of the DataSource
+    value: kubevirt-os-images
+  - name: CLOUD_USER_PASSWORD
+    description: Randomized password for the cloud-init user fedora
+    from: '[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{4}'
+    generate: expression
+```
+
+#### process subresource API
+
+A new` /process` subresource API will be implemented on
+`VirtualMachineTemplates` to allow simple server-side processing of templates.
+This is very much influenced by the OKD and OpenShift implementation of the
+`Template` CRD and will allow for easy processing of `VirtualMachineTemplates`
+through lightweight clients such as the console UI found on OKD and OpenShift.
+
+To provide this and other subresource APIs a `virt-template-api` service needs
+to be deployed and registered accordingly.
+
+#### create subresource API
+
+A new` /create` subresource API will be implemented on `VirtualMachineTemplates`
+to enable easy server-side creation of VMs from templates. By adding this
+subresource API, cluster admins can restrict users via RBAC to only create VMs
+from approved templates.
+
+#### virtctl process ${VirtualMachineTemplateName}
+
+A new process command will be introduced to the `virtctl` binary, again very
+much influenced by the `oc process` command provided downstream by OKD and
+OpenShift.
+
+This command processes a `VirtualMachineTemplate`, which can be specified as a
+reference to a cluster object or a local file. It applies any supplied
+parameters and outputs the rendered `VirtualMachine` manifest to standard
+output (stdout).
+
+```shell
+$ virtctl process fedora -p foo=bar
+```
+
+Internally the` process` command should use the same processing logic as the
+`/process` subresource API that is described above. This should be
+achieved by vendoring the logic that is hosted in the external
+`kubevirt/virt-template` repository into `kubevirt/kubevirt` for use within
+`virtctl`.
+
+#### VirtualMachineTemplateRequest
+
+A `VirtualMachineTemplateRequest` CRD will be introduced in the
+`template.kubevirt.io` APIGroup. This CRD will allow for an existing
+VirtualMachine to be captured as a template.
+
+This process will be orchestrated by a new controller that processes
+`VirtualMachineTemplateRequests`, captures a `VirtualMachineSnapshot` of the
+target `VirtualMachine`, clones this snapshot via a `DataVolume` into the target
```

Comment:
this new controller actually creates DataVolumes that may refer to sources in a different namespace? How do you validate that the user creating the `VirtualMachineTemplateRequest` has appropriate permission?